src/main/java/com/amazonaws/services/kinesis/connectors/KinesisConnectorRecordProcessor.java

```diff
@@ -126,8 +126,18 @@ public void processRecords(List<Record> records, IRecordProcessorCheckpointer ch
                 } else {
                     throw new RuntimeException("Transformer must implement ITransformer or ICollectionTransformer");
                 }
-            } catch (IOException e) {
+            } catch (Exception e) {
                 LOG.error(e);
+                try {
+                    final String lastSequenceNumberProcessed = buffer.getLastSequenceNumber();
```

Comment:
I don't like the idea of throwing those records away, even if they're logged.  Recovering the bad records from the logs can be difficult or impossible in some environments.  Still, it's probably better than losing records that expire in the Kinesis stream while your app loops endlessly.

I took a different approach.  The transformer in my application produces two different classes with a common parent class.  One subclass is for valid records, and the other is for those that caused problems with the parser.  Then the emitter separates the records and emits them to two different destinations.  It would be nice if the connectors library made this approach easier, but it still works well.